docs/data/data-grid/server-side-data/ServerSideLazyLoading.js

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import { DataGridPro, GridToolbar } from '@mui/x-data-grid-pro';
+import { useMockServer } from '@mui/x-data-grid-generator';
+
+function ServerSideLazyLoading() {
+  const { columns, fetchRows } = useMockServer(
+    { rowLength: 120 },
+    { useCursorPagination: false, minDelay: 300, maxDelay: 800 },
+  );
+
+  const dataSource = React.useMemo(
+    () => ({
+      getRows: async (params) => {
+        const urlParams = new URLSearchParams({
+          filterModel: encodeURIComponent(JSON.stringify(params.filterModel)),
+          sortModel: encodeURIComponent(JSON.stringify(params.sortModel)),
```

Comment:
```suggestion
          sortModel: JSON.stringify(params.sortModel),
```

🤔 The `URLSearchParams` should take care of encoding. This would double encode the query values. 
```tsx
String(new URLSearchParams({ foo: '/' }));
// 'foo=%2F'

// vs.

String(new URLSearchParams({ foo: encodeURIComponent('/') })); 
// 'foo=%252F'
```
Does the server do a double decode? Perhaps that's a bug?

Reply:
Thank you both 😄 
this is being updated now